### DISCUSSION TITLE
How many TimeZones should Watchy GSR be capable of handling?

### DISCUSSION THREAD

**Discussion by GuruSR** [GuruSR/Watchy_GSR#15] [Polls]
The actual code will be in Olson2POSIX, but Watchy GSR will have to handle them, so the feature will be to offer retrieving the UTC to Local based on the TimeZone index you want.  The index would be 0 (the default), 1 would be whatever you have it set for, 2, etc, retrieving the UTC to Local would go into a separate time_t value that you could then use to display that timezone.

I'd need to have functions to retrieve the information about that TimeZone (which will take some planning).

**Comment by JollyWizard** [comment]
My main concern is ease of setting the timezone.  I want a fast path to Up/Down the timezone offset and DST settings that doesn't require re-synchronization. 

But for US travel, 4 stored presets would be a logical choice.  That way you can go from one coast to the other with simple picks.

It would also be nice if the system was designed so that it allows seamless port of existing faces, so that the current timezone settings affect the reported time from existing code, rather than have to have to modify them for another API.  

If that can't be done, it would be nice to improve the porting of existing faces in a way that is already config aware.  In the Mario face, each digit is rendered separately inside a graphic and it is not 12/24 aware.  Providing the character information at the specific digit level would be ideal for this scenario.  12/24 would ideally be accounted for when retrieving the digits.  A sugar structure that accounts for this and timezone settings would be great.

Being able to label time zone settings via web config would also be a nice way to improve faces and configuration readability.

**Reply by GuruSR** [reply]
SmallRTC receives UTC from Watchy GSR, while each and every Clock update that happens, it converts the current UTC to localTime based on the TimeZone variable, so as long as the TZ is what is wanted, the Clock update just updates that time.  I'd merely have to add the ability to "Update the clock for X timezone", X being the timezone # in the list.  It won't be a massive issue to add them in, plus the OTA Website can be used on a laptop, tablet, phone to pick the TimeZones for each entry out of the dropdown list.

No re-sync needed which saves on battery.  The labeling of the zones would be something I'd definitely put in.
